packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4245,7 +4243,9 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
 
   void _updateInheritance() {
     assert(_lifecycleState == _ElementLifecycle.active);
-    _inheritedWidgets = _parent?._inheritedWidgets;
+    _inheritedAncestor = _parent is InheritedElement?
```

Comment:
You can avoid a type check by adding a private getter to Element and InheritedElement.

Element.getInheritedAncestor returns the value of the field
InheritedElement.getInheritedAncestor return itself

Reply:
How would I set the value of the field though?

Reply:
Added a polymorphic getter for `_thisOrParentAsInherited`, which `InheritedElement` overrides to return `this`.